src/test/java/io/cryostat/agent/FlightRecorderHelperTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.cryostat.agent.FlightRecorderHelper.TemplatedRecording;
+
+import jdk.jfr.Recording;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class FlightRecorderHelperTest {
+    @ParameterizedTest
+    @ValueSource(strings = {"default", "Continuous", "profile", "Profiling", "ALL"})
+    public void testCreateRecordingWithTemplate(String template) {
+        FlightRecorderHelper helper = new FlightRecorderHelper();
+
+        Optional<TemplatedRecording> recording =
+                helper.createRecordingWithPredefinedTemplate(template);
+
+        switch (template) {
+            case "ALL":
+                assertNotNull(recording);
```

Comment:
Actually, hang on - I think all of these cases should behave the same - the `Optional recording` should be not null and not empty for all of these `@ParameterizedTest` values.

Reply:
I think a better approach is to remove the switch statement and perform common assertions for all template `Label` or `name` values.